File: packages/react-native-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/Media.java
package org.wordpress.mobile.WPAndroidGlue;

import androidx.annotation.NonNull;

import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.MediaType;
import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.RNMedia;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.bridge.WritableNativeMap;

import java.util.Locale;

public class Media implements RNMedia {
    private int mId;
    private String mUrl;
    private String mType;
    private String mCaption;

    public Media(int id, String url, String type, String caption) {
        this.mId = id;
        this.mUrl = url;
        this.mType = type;
        this.mCaption = caption;
    }

    public Media(int id, String url) {
        this.mId = id;
        this.mUrl = url;
        this.mType = "";
        this.mCaption = "";
    }

    public static Media createRNMediaUsingMimeType(final int id, final String url, @NonNull final String mimeType, final String caption) {
        String type;

        if (mimeType.startsWith(MediaType.IMAGE.name().toLowerCase(Locale.ROOT))) {
            type =  MediaType.IMAGE.name().toLowerCase(Locale.ROOT);
        } else if (mimeType.startsWith(MediaType.VIDEO.name().toLowerCase(Locale.ROOT))) {
            type =  MediaType.VIDEO.name().toLowerCase(Locale.ROOT);
        } else {
            type = MediaType.OTHER.name().toLowerCase(Locale.ROOT);
        }

        return new Media(id, url, type, caption);
    }


    public int getId() {
        return mId;
    }

    public void setId(int id) {
        this.mId = id;
    }

    public String getUrl() {
        return mUrl;
    }

    public void setUrl(String url) {
        this.mUrl = url;
    }

    public String getType() {
        return mType;
    }

    public void setType(String mediaType) {
        this.mType = mediaType;
    }

    public String getCaption() {
        return mCaption;
    }

    public void setCaption(String caption) {
        this.mCaption = caption;
    }

    public WritableMap toMap() {
        WritableMap map = new WritableNativeMap();
        map.putInt("id", mId);
        map.putString("url", mUrl);
        map.putString("type", mType);
        map.putString("caption", mCaption);
        return map;
    }
}

File: packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/WPAndroidGlue/Media.kt
package org.wordpress.mobile.WPAndroidGlue

import com.facebook.react.bridge.WritableMap
import com.facebook.react.bridge.WritableNativeMap
import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.MediaType
import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.MediaType.IMAGE
import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.MediaType.OTHER
import org.wordpress.mobile.ReactNativeGutenbergBridge.GutenbergBridgeJS2Parent.MediaType.VIDEO
import org.wordpress.mobile.ReactNativeGutenbergBridge.RNMedia
import java.util.Locale

data class Media(
    override val id: Int,
    override val url: String,
    override val type: String,
    override val caption: String = "",
    override val title: String = "",
    override val alt: String = "",
    override val metadata: WritableMap = WritableNativeMap()
) : RNMedia {
    override fun toMap(): WritableMap = WritableNativeMap().apply {
        putInt("id", id)
        putString("url", url)
        putString("type", type)
        putString("caption", caption)
        putString("title", title)
        putString("alt", alt)
        putMap("metadata", metadata)
    }

    companion object {
        private fun convertToType(mimeType: String?): String {
            val isMediaType = { mediaType: MediaType ->
                mimeType?.startsWith(mediaType.name.toLowerCase(Locale.ROOT)) == true
            }
            val type = when {
                isMediaType(IMAGE) -> IMAGE
                isMediaType(VIDEO) -> VIDEO
                else -> OTHER
            }.name.toLowerCase(Locale.ROOT)
            return type;
        }

        @JvmStatic
        fun createRNMediaUsingMimeType(
            id: Int,
            url: String,
            mimeType: String?,
            caption: String?,
            title: String?,
            alt: String?,
        ): Media {
            val type = convertToType(mimeType)
            return Media(id, url, type, caption ?: "", title ?: "", alt ?: "")
        }
        @JvmStatic
        fun createRNMediaUsingMimeType(
            id: Int,
            url: String,
            mimeType: String?,
            caption: String?,
            title: String?,
            alt: String?,
            metadata: WritableNativeMap = WritableNativeMap()
        ): Media {
            val type = convertToType(mimeType)
            return Media(id, url, type, caption ?: "", title ?: "", alt ?: "", metadata)
        }
    }
}
